### ISSUE TITLE
pod install: "target has transitive dependencies that include static binaries"

### ISSUE THREAD

**Issue by ldiqual** [AppsFlyerSDK/segment-appsflyer-ios#3]
When trying to add 

```
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '9.0'
use_frameworks! # Because of swift
pod 'segment-appsflyer-ios', '1.1.1'
```

and running `pod install`, I get the following error:

```
[!] The '<mytarget>' target has transitive dependencies that include static binaries: (/path/to/pods/AppsFlyerFramework/AppsFlyer.framework)
```

which seems to be due to this: https://github.com/CocoaPods/CocoaPods/issues/2926

I was able to workaround the error by adding [this](https://github.com/CocoaPods/CocoaPods/issues/2926#issuecomment-136766631) to my Podfile:

```
pre_install do |installer|
    # workaround for https://github.com/CocoaPods/CocoaPods/issues/3289
    def installer.verify_no_static_framework_transitive_dependencies; end
end
```

However after the workaround I get this new error when compiling:

```
#import <AppsFlyer/AppsFlyer.h> // /path/to/pods/segment-appsflyer-ios/segment-appsflyer-ios/Classes/SEGAppsFlyerIntegration.h:11:9: 'AppsFlyer/AppsFlyer.h' file not found
```

I'm using cocoapods 1.0.1 and targeting iOS 9 only.

Any workaround?

**Comment by BrendaAF** [comment]
The current work around is to copy the source code for the integration into the app manually. 
It looks like Segment users have run into a similar issue previously with other partners. [Here](https://segment.com/docs/integrations/google-analytics/#target-has-transitive-dependencies-that-include-static-binaries) is documentation from Segment for this workaround.

**Comment by ldiqual** [comment]
@BrendaAF Unfortunately this is not something we're ready to do. It makes dependency version tracking + update really hard, in addition of polluting the app's project with numerous compiler flags and subdependencies. I believe that the ideal way to fix it is to provide an AppsFlyer framework without static binaries, and if I understand this correctly it would require AppsFlyer to provide an open-source version of the framework. Is it something that's in your pipeline?

What would be required on our end to integrate AppsFlyer *without* segment? Assuming we don't need event tracking, only ad attribution.

**Comment by BrendaAF** [comment]
@ldiqual 
[Here](https://support.appsflyer.com/hc/en-us/articles/207032126-AppsFlyer-SDK-Integration-Android) is the documentation around how to integrate AppsFlyer's native SDK.

**Comment by af-golan** [comment]
Fixed, released a new pod 1.1.4

**Comment by f2prateek** [comment]
@ldiqual I just saw your comment and wanted to chime in https://github.com/AppsFlyerSDK/segment-appsflyer-ios/issues/3#issuecomment-244176513.

I'm not totally sure why you think copying the integration code update approach would make version tracking any harder than natively installing the SDKs.

To be clear, the workaround suggested was to copy the source files from this repo (https://github.com/AppsFlyerSDK/segment-appsflyer-ios/tree/master/segment-appsflyer-ios/Classes) into your project. You'd still use cocoapods to integrate the AppsFlyer and Segment SDK's like this:

```
pod "Analytics"
pod "AppsFlyer"
...
```

This is a workaround because instead of a transitive dependency (segment-applslyfer-ios -> AppsFlyer), you depend on AppsFlyer *directly*, and copy the "mapping" (i.e. this integration) code into your app.

You'd be missing out on using the integration via Cocoapods (which is just this repo, not the core AppsFlyer SDK or Analytics SDK), and while it's not ideal, I think it's better than integrating the tools natively and having to write your mapping layers from scratch.
